buildkite/resource_team_member.go

```diff
@@ -3,174 +3,208 @@ package buildkite
 )
 
-type TeamMemberNode struct {
-	ID   graphql.String
-	Role TeamMemberRole
-	UUID graphql.String
-	Team TeamNode
-	User struct {
-		ID graphql.ID
-	}
+type teamMemberResourceModel struct {
+	Id     types.String `tfsdk:"id"`
+	Uuid   types.String `tfsdk:"uuid"`
+	Role   types.String `tfsdk:"role"`
+	TeamId types.String `tfsdk:"team_id"`
+	UserId types.String `tfsdk:"user_id"`
 }
 
-func resourceTeamMember() *schema.Resource {
-	return &schema.Resource{
-		CreateContext: CreateTeamMember,
-		ReadContext:   ReadTeamMember,
-		UpdateContext: UpdateTeamMember,
-		DeleteContext: DeleteTeamMember,
-		Importer: &schema.ResourceImporter{
-			State: schema.ImportStatePassthrough,
-		},
+type teamMemberResource struct {
+	client *Client
+}
 
-		Schema: map[string]*schema.Schema{
-			"role": &schema.Schema{
-				Required: true,
-				Type:     schema.TypeString,
-				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
-					v := val.(string)
-					switch v {
-					case "MEMBER":
-					case "MAINTAINER":
-						return
-					default:
-						errs = append(errs, fmt.Errorf("%q must be either MEMBER or MAINTAINER, got: %s", key, v))
-						return
-					}
-					return
+func newTeamMemberResource() resource.Resource {
+	return &teamMemberResource{}
+}
+
+func (teamMemberResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_member"
+}
+
+func (tm *teamMemberResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	tm.client = req.ProviderData.(*Client)
+}
+
+func (teamMemberResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_schema.Schema{
+		MarkdownDescription: "A team member resourc allows for the management of team team membership for existing organization users.",
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
-			"team_id": &schema.Schema{
-				Required: true,
-				Type:     schema.TypeString,
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"user_id": &schema.Schema{
-				Required: true,
-				Type:     schema.TypeString,
+			"team_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the team to add/remove the user to/from.",
 			},
-			"uuid": &schema.Schema{
-				Computed: true,
-				Type:     schema.TypeString,
+			"user_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the user to add/remove.",
```

Comment:
Same. Would simply not providing a `user_id` remove that user from the team? When setting this value, I don't think I set it if I want to `remove`.

Reply:
Yeah removing this resource removes them from the team; it removes the link between user and team. So the descriptions here are simply to "add"

Reply:
Adjusting these - thanks both